### PR TITLE
Update lapack to 3.6.0

### PR DESCRIPTION
**Pull request by jschueller**

### CODE REVIEW

File: lapack/build.sh [hunk 1]
```diff
@@ -2,10 +2,11 @@ mkdir build
...
 
 cmake \
-    -DCMAKE_INSTALL_PREFIX=$PREFIX \
-    -DBUILD_TESTING=OFF \
-    -DBUILD_SHARED_LIBS=ON \
-    ..
+  -DCMAKE_INSTALL_PREFIX=$PREFIX \
+  -DBUILD_TESTING=OFF \
+  -DBUILD_SHARED_LIBS=ON \
+  -DLAPACKE=ON \
```
Comment (ccordoba12) [inline lapack/build.sh]: What is this new option used for?

Comment (jakirkham) [inline lapack/build.sh]: I believe this is for the C interface to LAPACK. Am a bit surprised this wasn't already enabled.

**Comment by jschueller** [comment]
@ccordoba12 
This enables lapacke, the c bindings for lapack, see http://www.netlib.org/lapack/lapacke.html.
The next OpenTURNS version is able to use it instead of needing a fortran compiler, so I enabled it.

**Comment by jakirkham** [comment]
Includes https://github.com/conda/conda-recipes/pull/411

**Comment by jschueller** [comment]
@jakirkham, unfortunately the previous 3.5.0 version had no cmake config files and so it could not be automagically detected out of the box, so we did not use it.

**Comment by ccordoba12** [comment]
Ok, thanks for the clarifications :-) Merging!

**Pull request merged**